### PR TITLE
chore: upgrade to redux-toolkit 1.8

### PR DESCRIPTION
**Pull request by justindomingue**
-mostly to get this fix from rtk-query: https://github.com/reduxjs/redux-toolkit/issues/1339

for different hook args, `data` returned by `useQuery` was the same. To go around this behavior in the interface, we used a very ugly hack to detect when the `Trade` input/output were different from hook input/output. This is not needed anymore thanks to `currentData`

also fixed an existing bug with priceImpact relying on swap form inputs, rather than resolved Trade.input/output. Since we debounce the trade, price impact would often lag behind.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -210,7 +210,7 @@
     "@fontsource/ibm-plex-mono": "^4.5.1",
     "@fontsource/inter": "^4.5.1",
     "@popperjs/core": "^2.4.4",
-    "@reduxjs/toolkit": "^1.6.1",
+    "@reduxjs/toolkit": "^1.8.0",
```
[the remark this answers is not in the document]
Comment (justindomingue) [reply package.json:213]: I think it's fine to leave as 1.6.1.

**Review by snreynolds — APPROVED** [review]

File: src/hooks/useBestTrade.ts [hunk 2]
```diff
@@ -36,21 +36,8 @@ export function useBestTrade(
...
-  const debouncing =
-    routingAPITrade.trade &&
-    amountSpecified &&
-    (tradeType === TradeType.EXACT_INPUT
-      ? !routingAPITrade.trade.inputAmount.equalTo(amountSpecified) ||
-        !amountSpecified.currency.equals(routingAPITrade.trade.inputAmount.currency) ||
-        !debouncedOtherCurrency?.equals(routingAPITrade.trade.outputAmount.currency)
-      : !routingAPITrade.trade.outputAmount.equalTo(amountSpecified) ||
-        !amountSpecified.currency.equals(routingAPITrade.trade.outputAmount.currency) ||
-        !debouncedOtherCurrency?.equals(routingAPITrade.trade.inputAmount.currency))
```
Comment (JFrankfurt) [inline src/hooks/useBestTrade.ts:51]: tyvm for nuking this block <3

**Review by JFrankfurt — APPROVED** [review]
what's the reason for the Token→Currency change on useUSDCValue?

**Comment by justindomingue** [comment]
@JFrankfurt I also fixed an existing bug with `priceImpact` relying on swap form inputs, rather than resolved `Trade.input/output`. Since we debounce the trade, price impact would often lag behind.

**Pull request merged**